kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/FeedbackInputFragment.java

```diff
@@ -0,0 +1,245 @@
+        View view = inflater.inflate(R.layout.feedback_rating_layout, container, false);
+
+        editTextFeedbackComment = view.findViewById(R.id.idEditTextFeedback);
+        buttonSubmitFeedback = view.findViewById(R.id.idButtonSubmit);
+
+        buttonCloseFragment = view.findViewById(R.id.idCloseFeedbackFragment);
+
+        editTextFeedbackComment.setVisibility(View.GONE);
+
+        //selected
+        Drawable[] buttonDrawablesSelected = new Drawable[3];
+        buttonDrawablesSelected[0] = getResources().getDrawable(R.drawable.ic_sad_1);
+        buttonDrawablesSelected[1] = getResources().getDrawable(R.drawable.ic_confused);
+        buttonDrawablesSelected[2] = getResources().getDrawable(R.drawable.ic_happy);
+        //not selected
+        Drawable[] buttonDrawablesUnselected = new Drawable[3];
+        buttonDrawablesUnselected[0] = getResources().getDrawable(R.drawable.ic_sad_1_grey);
+        buttonDrawablesUnselected[1] = getResources().getDrawable(R.drawable.ic_confused_grey);
+        buttonDrawablesUnselected[2] = getResources().getDrawable(R.drawable.ic_happy_grey);
+
+        feedbackRatingGroup = new FeedbackRatingGroup(3, buttonDrawablesSelected, buttonDrawablesUnselected);
+
+        feedbackRatingGroup.createViewForRatingValue(view, 1, R.id.idButtonPoor, R.id.idTextPoor);
+        feedbackRatingGroup.createViewForRatingValue(view, 2, R.id.idButtonAverage, R.id.idTextAverage);
+        feedbackRatingGroup.createViewForRatingValue(view, 3, R.id.idButtonGood, R.id.idTextGood);
+
+        buttonSubmitFeedback.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(ratingLevel == 0) {
+                    setRatingLevel(RATING_AVERAGE);
+                }
+                String feedbackComment = editTextFeedbackComment.getText().toString().trim();
+                feedbackFragmentListener.onFeedbackSubmitButtonPressed(getRatingLevel(), feedbackComment);
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        buttonCloseFragment.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        return view;
+    }
+
+    /**
+     * toggle button as selected or deselected
+     * @param select true or false
+     * @param feedbackRating the views to toggle
+     */
+    void toggleRatingButtonSelected(boolean select, FeedbackRatingGroup.FeedbackRating feedbackRating) {
+        //show the feedback comment input edit text, if not already visible
+        if(editTextFeedbackComment.getVisibility()==View.GONE) {
+            editTextFeedbackComment.setVisibility(View.VISIBLE);
+        }
+
+        //set drawable
+        feedbackRatingGroup.setDrawableForValue(select, feedbackRating.ratingValue);
```

Comment:
is this method setting values on the same 'feedbackRating' object or different?
if it is the same then do we need to fetch the same object again from value which we already have here?

Reply:
passing FeedbackRating object now